s390x test failures

The Debian build logs at https://buildd.debian.org/status/package.php?p=gnuradio show that many tests currently fail on the s390x architecture. Many of the tests exit with segmentation faults, and many others fail with what appear to be endianness bugs.

Although it's unlikely that anyone is running GNU Radio on this architecture, the test failures very likely point to places where GNU Radio depends on undefined behaviour. These bugs could well lead to failures on other architectures in the future, so I think they should be fixed.

Here's a full log of the test run from 3.10.4.0-3 on the s390x architecture:
```
        Start   1: runtime_qa_buffer.cc
  1/261 Test   #1: runtime_qa_buffer.cc .........................   Passed    1.84 sec
        Start   2: runtime_qa_io_signature.cc
  2/261 Test   #2: runtime_qa_io_signature.cc ...................   Passed    0.03 sec
        Start   3: runtime_qa_logger.cc
  3/261 Test   #3: runtime_qa_logger.cc .........................   Passed    0.02 sec
        Start   4: runtime_qa_host_buffer.cc
  4/261 Test   #4: runtime_qa_host_buffer.cc ....................   Passed    0.04 sec
        Start   5: runtime_qa_vmcircbuf.cc
  5/261 Test   #5: runtime_qa_vmcircbuf.cc ......................   Passed    0.10 sec
        Start   6: math_qa_fxpt.cc
  6/261 Test   #6: math_qa_fxpt.cc ..............................   Passed    0.03 sec
        Start   7: math_qa_fxpt_nco.cc
  7/261 Test   #7: math_qa_fxpt_nco.cc ..........................   Passed    0.14 sec
        Start   8: math_qa_fxpt_vco.cc
  8/261 Test   #8: math_qa_fxpt_vco.cc ..........................   Passed    0.13 sec
        Start   9: math_qa_math.cc
  9/261 Test   #9: math_qa_math.cc ..............................   Passed    0.01 sec
        Start  10: math_qa_sincos.cc
 10/261 Test  #10: math_qa_sincos.cc ............................   Passed    0.01 sec
        Start  11: math_qa_fast_atan2f.cc
 11/261 Test  #11: math_qa_fast_atan2f.cc .......................   Passed    0.03 sec
        Start  12: pmt_prims
 12/261 Test  #12: pmt_prims ....................................   Passed    0.49 sec
        Start  13: pmt_unv
 13/261 Test  #13: pmt_unv ......................................   Passed    0.02 sec
        Start  14: qa_flowgraph
 14/261 Test  #14: qa_flowgraph .................................   Passed    0.46 sec
        Start  15: qa_prefs
 15/261 Test  #15: qa_prefs .....................................   Passed    0.12 sec
        Start  16: qa_random
 16/261 Test  #16: qa_random ....................................   Passed    0.45 sec
        Start  17: qa_tag_utils
 17/261 Test  #17: qa_tag_utils .................................   Passed    0.12 sec
        Start  18: qa_hier_block2
 18/261 Test  #18: qa_hier_block2 ...............................   Passed    2.28 sec
        Start  19: qa_uncaught_exception
 19/261 Test  #19: qa_uncaught_exception ........................   Passed    2.69 sec
        Start  20: qa_pmt
 20/261 Test  #20: qa_pmt .......................................***Failed    0.11 sec
......................F
======================================================================
FAIL: test23_absolute_serialization_float_uvecs (__main__.test_pmt)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<<PKGBUILDDIR>>/gnuradio-runtime/python/pmt/qa_pmt.py", line 350, in test23_absolute_serialization_float_uvecs
    self.assertEqual(out_str, in_str)
AssertionError: b'\n\n\x00\x00\x00\x02\x01\x00K\x81\x01\x82\xbf\x80\x00\x00@I\x0eVB\xc7\xfa\xe1' != b'\n\n\x00\x00\x00\x02\x01\x00\xbf\x80\x00\x00K\x81\x01\x82B\xc7\xfa\xe1@I\x0eV'

----------------------------------------------------------------------
Ran 23 tests in 0.001s

FAILED (failures=1)
((a . 123765))

        Start  21: qa_pmt_to_python
 21/261 Test  #21: qa_pmt_to_python .............................   Passed    0.11 sec
        Start  22: blocks_qa_block_tags.cc
 22/261 Test  #22: blocks_qa_block_tags.cc ......................   Passed    0.04 sec
        Start  23: blocks_qa_gr_block.cc
 23/261 Test  #23: blocks_qa_gr_block.cc ........................   Passed    0.03 sec
        Start  24: blocks_qa_gr_flowgraph.cc
 24/261 Test  #24: blocks_qa_gr_flowgraph.cc ....................   Passed    0.03 sec
        Start  25: blocks_qa_gr_hier_block2.cc
 25/261 Test  #25: blocks_qa_gr_hier_block2.cc ..................   Passed    0.02 sec
        Start  26: blocks_qa_gr_hier_block2_derived.cc
 26/261 Test  #26: blocks_qa_gr_hier_block2_derived.cc ..........   Passed    0.03 sec
        Start  27: blocks_qa_gr_top_block.cc
 27/261 Test  #27: blocks_qa_gr_top_block.cc ....................   Passed    0.04 sec
        Start  28: blocks_qa_rotator.cc
 28/261 Test  #28: blocks_qa_rotator.cc .........................***Failed    0.05 sec
Running 2 test cases...
unknown location(0): [4;31;49mfatal error: in "t2": memory access violation at address: 0x3f7ffeb73bcd4000: no mapping at fault address[0;39;49m
./gr-blocks/lib/qa_rotator.cc(57): [1;36;49mlast checkpoint: "t2" test entry[0;39;49m

[1;31;49m*** 1 failure is detected in the test module "Master Test Suite"
[0;39;49m

        Start  29: blocks_qa_set_msg_handler.cc
 29/261 Test  #29: blocks_qa_set_msg_handler.cc .................   Passed    0.13 sec
        Start  30: qa_add_mult_div_sub
 30/261 Test  #30: qa_add_mult_div_sub ..........................***Failed    0.17 sec
..........Segmentation fault

        Start  31: qa_add_mult_v
 31/261 Test  #31: qa_add_mult_v ................................   Passed    0.20 sec
        Start  32: qa_affinity
 32/261 Test  #32: qa_affinity ..................................   Passed    0.14 sec
        Start  33: qa_argmax
 33/261 Test  #33: qa_argmax ....................................   Passed    0.14 sec
        Start  34: qa_block_behavior
 34/261 Test  #34: qa_block_behavior ............................   Passed    0.14 sec
        Start  35: qa_block_gateway
 35/261 Test  #35: qa_block_gateway .............................   Passed    0.23 sec
        Start  36: qa_blockinterleaver_xx
 36/261 Test  #36: qa_blockinterleaver_xx .......................   Passed    0.27 sec
        Start  37: qa_boolean_operators
 37/261 Test  #37: qa_boolean_operators .........................   Passed    0.16 sec
        Start  38: qa_burst_tagger
 38/261 Test  #38: qa_burst_tagger ..............................   Passed    0.14 sec
        Start  39: qa_conjugate
 39/261 Test  #39: qa_conjugate .................................   Passed    0.14 sec
        Start  40: qa_copy
 40/261 Test  #40: qa_copy ......................................   Passed    0.14 sec
        Start  41: qa_delay
 41/261 Test  #41: qa_delay .....................................   Passed    0.15 sec
        Start  42: qa_endian_swap
 42/261 Test  #42: qa_endian_swap ...............................   Passed    0.14 sec
        Start  43: qa_exponentiate_const_cci
 43/261 Test  #43: qa_exponentiate_const_cci ....................   Passed    0.30 sec
        Start  44: qa_file_descriptor_source_sink
 44/261 Test  #44: qa_file_descriptor_source_sink ...............   Passed    0.14 sec
        Start  45: qa_file_metadata
 45/261 Test  #45: qa_file_metadata .............................   Passed    0.15 sec
        Start  46: qa_file_sink
 46/261 Test  #46: qa_file_sink .................................   Passed    0.14 sec
        Start  47: qa_file_source
 47/261 Test  #47: qa_file_source ...............................   Passed    0.15 sec
        Start  48: qa_head
 48/261 Test  #48: qa_head ......................................   Passed    0.14 sec
        Start  49: qa_hier_block2
 49/261 Test  #49: qa_hier_block2 ...............................   Passed    2.22 sec
        Start  50: qa_hier_block2_message_connections
 50/261 Test  #50: qa_hier_block2_message_connections ...........   Passed    0.15 sec
        Start  51: qa_host_buffer_copy
 51/261 Test  #51: qa_host_buffer_copy ..........................   Passed    0.45 sec
        Start  52: qa_integrate
 52/261 Test  #52: qa_integrate .................................   Passed    0.15 sec
        Start  53: qa_interleave
 53/261 Test  #53: qa_interleave ................................   Passed    0.15 sec
        Start  54: qa_keep_m_in_n
 54/261 Test  #54: qa_keep_m_in_n ...............................   Passed    0.15 sec
        Start  55: qa_keep_one_in_n
 55/261 Test  #55: qa_keep_one_in_n .............................   Passed    0.14 sec
        Start  56: qa_logger
 56/261 Test  #56: qa_logger ....................................   Passed    0.14 sec
        Start  57: qa_matrix_interleaver
 57/261 Test  #57: qa_matrix_interleaver ........................   Passed    0.15 sec
        Start  58: qa_max
 58/261 Test  #58: qa_max .......................................   Passed    0.15 sec
        Start  59: qa_message
 59/261 Test  #59: qa_message ...................................   Passed    2.61 sec
        Start  60: qa_message_debug
 60/261 Test  #60: qa_message_debug .............................   Passed    3.52 sec
        Start  61: qa_message_strobe
 61/261 Test  #61: qa_message_strobe ............................   Passed    3.26 sec
        Start  62: qa_min
 62/261 Test  #62: qa_min .......................................   Passed    0.15 sec
        Start  63: qa_moving_average
 63/261 Test  #63: qa_moving_average ............................   Passed    0.19 sec
        Start  64: qa_multiply_conjugate
 64/261 Test  #64: qa_multiply_conjugate ........................   Passed    0.15 sec
        Start  65: qa_multiply_matrix_xx
 65/261 Test  #65: qa_multiply_matrix_xx ........................***Failed    0.25 sec
/<<PKGBUILDDIR>>/gr-blocks/python/blocks/qa_multiply_matrix_xx.py:55: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
  X_in = numpy.matrix(X_in)
/<<PKGBUILDDIR>>/gr-blocks/python/blocks/qa_multiply_matrix_xx.py:56: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
  A_matrix = numpy.matrix(A)
/<<PKGBUILDDIR>>/gr-blocks/python/blocks/qa_multiply_matrix_xx.py:60: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
  Y_out_exp = numpy.matrix(numpy.zeros((M, X_in.shape[1])))
/usr/lib/python3/dist-packages/numpy/matrixlib/defmatrix.py:69: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
  return matrix(data, dtype=dtype, copy=False)
.Segmentation fault

        Start  66: qa_mute
 66/261 Test  #66: qa_mute ......................................   Passed    0.16 sec
        Start  67: qa_nlog10
 67/261 Test  #67: qa_nlog10 ....................................   Passed    0.15 sec
        Start  68: qa_null_sink_source
 68/261 Test  #68: qa_null_sink_source ..........................   Passed    0.15 sec
        Start  69: qa_pack_k_bits
 69/261 Test  #69: qa_pack_k_bits ...............................   Passed    0.16 sec
        Start  70: qa_packed_to_unpacked
 70/261 Test  #70: qa_packed_to_unpacked ........................   Passed    0.21 sec
        Start  71: qa_patterned_interleaver
 71/261 Test  #71: qa_patterned_interleaver .....................   Passed    0.15 sec
        Start  72: qa_peak_detector
 72/261 Test  #72: qa_peak_detector .............................   Passed    0.15 sec
        Start  73: qa_peak_detector2
 73/261 Test  #73: qa_peak_detector2 ............................   Passed    0.15 sec
        Start  74: qa_pipe_fittings
 74/261 Test  #74: qa_pipe_fittings .............................   Passed    0.15 sec
        Start  75: qa_plateau_detector_fb
 75/261 Test  #75: qa_plateau_detector_fb .......................   Passed    0.15 sec
        Start  76: qa_probe_signal
 76/261 Test  #76: qa_probe_signal ..............................   Passed    0.34 sec
        Start  77: qa_python_message_passing
 77/261 Test  #77: qa_python_message_passing ....................   Passed    0.15 sec
        Start  78: qa_regenerate
 78/261 Test  #78: qa_regenerate ................................   Passed    0.15 sec
        Start  79: qa_repack_bits_bb
 79/261 Test  #79: qa_repack_bits_bb ............................   Passed    0.17 sec
        Start  80: qa_repeat
 80/261 Test  #80: qa_repeat ....................................   Passed    0.19 sec
        Start  81: qa_rms
 81/261 Test  #81: qa_rms .......................................   Passed    0.30 sec
        Start  82: qa_rotator_cc
 82/261 Test  #82: qa_rotator_cc ................................***Failed    0.18 sec
Segmentation fault

        Start  83: qa_sample_and_hold
 83/261 Test  #83: qa_sample_and_hold ...........................   Passed    0.17 sec
        Start  84: qa_selector
 84/261 Test  #84: qa_selector ..................................   Passed    0.19 sec
        Start  85: qa_skiphead
 85/261 Test  #85: qa_skiphead ..................................   Passed    0.23 sec
        Start  86: qa_stream_demux
 86/261 Test  #86: qa_stream_demux ..............................   Passed    0.20 sec
        Start  87: qa_stream_mux
 87/261 Test  #87: qa_stream_mux ................................   Passed    0.18 sec
        Start  88: qa_stream_to_tagged_stream
 88/261 Test  #88: qa_stream_to_tagged_stream ...................   Passed    0.16 sec
        Start  89: qa_stretch
 89/261 Test  #89: qa_stretch ...................................   Passed    0.18 sec
        Start  90: qa_tag_debug
 90/261 Test  #90: qa_tag_debug .................................   Passed    0.17 sec
        Start  91: qa_tag_file_sink
 91/261 Test  #91: qa_tag_file_sink .............................   Passed    0.16 sec
        Start  92: qa_tag_gate
 92/261 Test  #92: qa_tag_gate ..................................   Passed    0.17 sec
        Start  93: qa_tag_share
 93/261 Test  #93: qa_tag_share .................................   Passed    0.19 sec
        Start  94: qa_tagged_stream_mux
 94/261 Test  #94: qa_tagged_stream_mux .........................   Passed    0.29 sec
        Start  95: qa_tags_strobe
 95/261 Test  #95: qa_tags_strobe ...............................   Passed    0.17 sec
        Start  96: qa_threshold
 96/261 Test  #96: qa_threshold .................................   Passed    0.16 sec
        Start  97: qa_throttle
 97/261 Test  #97: qa_throttle ..................................   Passed    0.87 sec
        Start  98: qa_transcendental
 98/261 Test  #98: qa_transcendental ............................   Passed    0.14 sec
        Start  99: qa_tsb_vector_sink_X
 99/261 Test  #99: qa_tsb_vector_sink_X .........................   Passed    0.16 sec
        Start 100: qa_type_conversions
100/261 Test #100: qa_type_conversions ..........................   Passed    0.25 sec
        Start 101: qa_unpack_k_bits
101/261 Test #101: qa_unpack_k_bits .............................   Passed    0.19 sec
        Start 102: qa_vco
102/261 Test #102: qa_vco .......................................   Passed    0.19 sec
        Start 103: qa_vector_insert
103/261 Test #103: qa_vector_insert .............................   Passed    0.33 sec
        Start 104: qa_vector_map
104/261 Test #104: qa_vector_map ................................   Passed    0.22 sec
        Start 105: qa_vector_sink_source
105/261 Test #105: qa_vector_sink_source ........................   Passed    0.17 sec
        Start 106: qa_wavfile
106/261 Test #106: qa_wavfile ...................................***Failed    0.19 sec
..F.
======================================================================
FAIL: test_003_checkwav_append_copy (__main__.test_wavefile)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<<PKGBUILDDIR>>/gr-blocks/python/blocks/qa_wavfile.py", line 142, in test_003_checkwav_append_copy
    self.assertEqual(data_in_halved, data_out[2 * len(data_in):])
AssertionError: Lists differ: [96, 0, 32, 6418] != [224, 0, 32, -26351]

First differing element 0:
96
224

- [96, 0, 32, 6418]
+ [224, 0, 32, -26351]

----------------------------------------------------------------------
Ran 4 tests in 0.030s

FAILED (failures=1)

        Start 107: test_rs
107/261 Test #107: test_rs ......................................   Passed    0.02 sec
        Start 108: qa_ber_bf
108/261 Test #108: qa_ber_bf ....................................   Passed    0.36 sec
        Start 109: qa_depuncture
109/261 Test #109: qa_depuncture ................................   Passed    0.63 sec
        Start 110: qa_ecc_ccsds_27
110/261 Test #110: qa_ecc_ccsds_27 ..............................   Passed    0.18 sec
        Start 111: qa_fecapi_cc
111/261 Test #111: qa_fecapi_cc .................................   Passed    0.41 sec
        Start 112: qa_fecapi_cc_buffer_overflow
112/261 Test #112: qa_fecapi_cc_buffer_overflow .................   Passed    5.42 sec
        Start 113: qa_fecapi_dummy
113/261 Test #113: qa_fecapi_dummy ..............................   Passed    0.47 sec
        Start 114: qa_fecapi_ldpc
114/261 Test #114: qa_fecapi_ldpc ...............................   Passed    1.28 sec
        Start 115: qa_fecapi_repetition
115/261 Test #115: qa_fecapi_repetition .........................   Passed    0.61 sec
        Start 116: qa_polar_decoder_sc
116/261 Test #116: qa_polar_decoder_sc ..........................   Passed    0.62 sec
        Start 117: qa_polar_decoder_sc_list
117/261 Test #117: qa_polar_decoder_sc_list .....................   Passed    0.29 sec
        Start 118: qa_polar_decoder_sc_systematic
118/261 Test #118: qa_polar_decoder_sc_systematic ...............   Passed    0.26 sec
        Start 119: qa_polar_encoder
119/261 Test #119: qa_polar_encoder .............................   Passed    0.36 sec
        Start 120: qa_polar_encoder_systematic
120/261 Test #120: qa_polar_encoder_systematic ..................   Passed    0.35 sec
        Start 121: qa_puncture
121/261 Test #121: qa_puncture ..................................   Passed    0.91 sec
        Start 122: fft_qa_fft_shift.cc
122/261 Test #122: fft_qa_fft_shift.cc ..........................   Passed    0.01 sec
        Start 123: qa_fft
123/261 Test #123: qa_fft .......................................   Passed    0.19 sec
        Start 124: qa_goertzel
124/261 Test #124: qa_goertzel ..................................   Passed    0.15 sec
        Start 125: qa_window
125/261 Test #125: qa_window ....................................   Passed    0.12 sec
        Start 126: filter_qa_firdes.cc
126/261 Test #126: filter_qa_firdes.cc ..........................   Passed    0.03 sec
        Start 127: filter_qa_fir_filter_with_buffer.cc
127/261 Test #127: filter_qa_fir_filter_with_buffer.cc ..........   Passed    0.09 sec
        Start 128: filter_qa_mmse_fir_interpolator_cc.cc
128/261 Test #128: filter_qa_mmse_fir_interpolator_cc.cc ........   Passed    0.03 sec
        Start 129: filter_qa_mmse_fir_interpolator_ff.cc
129/261 Test #129: filter_qa_mmse_fir_interpolator_ff.cc ........   Passed    0.03 sec
        Start 130: filter_qa_mmse_interp_differentiator_cc.cc
130/261 Test #130: filter_qa_mmse_interp_differentiator_cc.cc ...   Passed    0.04 sec
        Start 131: filter_qa_mmse_interp_differentiator_ff.cc
131/261 Test #131: filter_qa_mmse_interp_differentiator_ff.cc ...   Passed    0.03 sec
        Start 132: qa_dc_blocker
132/261 Test #132: qa_dc_blocker ................................   Passed    0.16 sec
        Start 133: qa_fft_filter
133/261 Test #133: qa_fft_filter ................................   Passed    1.39 sec
        Start 134: qa_file_taps_loader
134/261 Test #134: qa_file_taps_loader ..........................   Passed    0.16 sec
        Start 135: qa_filter_delay_fc
135/261 Test #135: qa_filter_delay_fc ...........................   Passed    0.16 sec
        Start 136: qa_filterbank
136/261 Test #136: qa_filterbank ................................   Passed    0.16 sec
        Start 137: qa_fir_filter
137/261 Test #137: qa_fir_filter ................................   Passed    0.17 sec
        Start 138: qa_firdes
138/261 Test #138: qa_firdes ....................................   Passed    0.15 sec
        Start 139: qa_fractional_interpolator
139/261 Test #139: qa_fractional_interpolator ...................   Passed    0.18 sec
        Start 140: qa_fractional_resampler
140/261 Test #140: qa_fractional_resampler ......................   Passed    0.21 sec
        Start 141: qa_freq_xlating_fft_filter
141/261 Test #141: qa_freq_xlating_fft_filter ...................***Failed    0.16 sec
Segmentation fault

        Start 142: qa_freq_xlating_fir_filter
142/261 Test #142: qa_freq_xlating_fir_filter ...................***Failed    0.14 sec
Segmentation fault

        Start 143: qa_hilbert
143/261 Test #143: qa_hilbert ...................................   Passed    0.15 sec
        Start 144: qa_iir_filter
144/261 Test #144: qa_iir_filter ................................   Passed    0.17 sec
        Start 145: qa_interp_fir_filter
145/261 Test #145: qa_interp_fir_filter .........................   Passed    0.15 sec
        Start 146: qa_pfb_arb_resampler
146/261 Test #146: qa_pfb_arb_resampler .........................   Passed    0.32 sec
        Start 147: qa_pfb_channelizer
147/261 Test #147: qa_pfb_channelizer ...........................   Passed    0.36 sec
        Start 148: qa_pfb_decimator
148/261 Test #148: qa_pfb_decimator .............................   Passed    0.54 sec
        Start 149: qa_pfb_interpolator
149/261 Test #149: qa_pfb_interpolator ..........................   Passed    0.17 sec
        Start 150: qa_pfb_synthesizer
150/261 Test #150: qa_pfb_synthesizer ...........................   Passed    0.33 sec
        Start 151: qa_pm_remez
151/261 Test #151: qa_pm_remez ..................................   Passed    0.15 sec
        Start 152: qa_rational_resampler
152/261 Test #152: qa_rational_resampler ........................   Passed    0.18 sec
        Start 153: qa_single_pole_iir
153/261 Test #153: qa_single_pole_iir ...........................   Passed    0.16 sec
        Start 154: qa_agc
154/261 Test #154: qa_agc .......................................   Passed    0.17 sec
        Start 155: qa_cpfsk
155/261 Test #155: qa_cpfsk .....................................   Passed    0.16 sec
        Start 156: qa_ctcss_squelch
156/261 Test #156: qa_ctcss_squelch .............................   Passed    0.16 sec
        Start 157: qa_dpll
157/261 Test #157: qa_dpll ......................................   Passed    0.16 sec
        Start 158: qa_fastnoise
158/261 Test #158: qa_fastnoise .................................   Passed    7.81 sec
        Start 159: qa_fm_emph
159/261 Test #159: qa_fm_emph ...................................   Passed    0.17 sec
        Start 160: qa_fmdet
160/261 Test #160: qa_fmdet .....................................   Passed    0.15 sec
        Start 161: qa_frequency_modulator
161/261 Test #161: qa_frequency_modulator .......................   Passed    0.16 sec
        Start 162: qa_noise
162/261 Test #162: qa_noise .....................................   Passed    0.15 sec
        Start 163: qa_phase_modulator
163/261 Test #163: qa_phase_modulator ...........................   Passed    0.16 sec
        Start 164: qa_pll_carriertracking
164/261 Test #164: qa_pll_carriertracking .......................   Passed    0.16 sec
        Start 165: qa_pll_freqdet
165/261 Test #165: qa_pll_freqdet ...............................   Passed    0.16 sec
        Start 166: qa_pll_refout
166/261 Test #166: qa_pll_refout ................................   Passed    0.16 sec
        Start 167: qa_probe_avg_mag_sqrd
167/261 Test #167: qa_probe_avg_mag_sqrd ........................   Passed    0.16 sec
        Start 168: qa_pwr_squelch
168/261 Test #168: qa_pwr_squelch ...............................   Passed    0.16 sec
        Start 169: qa_quadrature_demod
169/261 Test #169: qa_quadrature_demod ..........................   Passed    0.16 sec
        Start 170: qa_rail_ff
170/261 Test #170: qa_rail_ff ...................................   Passed    0.16 sec
        Start 171: qa_random_uniform_source
171/261 Test #171: qa_random_uniform_source .....................   Passed    0.17 sec
        Start 172: qa_sig_source
172/261 Test #172: qa_sig_source ................................   Passed    0.17 sec
        Start 173: qa_simple_squelch
173/261 Test #173: qa_simple_squelch ............................   Passed    0.16 sec
        Start 174: digital_qa_header_format.cc
174/261 Test #174: digital_qa_header_format.cc ..................***Failed    0.04 sec
Running 4 test cases...
./gr-digital/lib/qa_header_format.cc(71): [4;31;49mfatal error: in "test_default_format": critical check 0xAA == (int)h0 has failed [170 != 0][0;39;49m
./gr-digital/lib/qa_header_format.cc(154): [4;31;49mfatal error: in "test_counter_format": critical check 0xAA == (int)h0 has failed [170 != 0][0;39;49m

[1;31;49m*** 2 failures are detected in the test module "Master Test Suite"
[0;39;49m

        Start 175: digital_qa_header_buffer.cc
175/261 Test #175: digital_qa_header_buffer.cc ..................***Failed    0.04 sec
Running 10 test cases...
./gr-digital/lib/qa_header_buffer.cc(48): [4;31;49mfatal error: in "test_add16": critical check (uint8_t)0xAF == header.header()[0] has failed [0xaf != 0x5c][0;39;49m
./gr-digital/lib/qa_header_buffer.cc(87): [4;31;49mfatal error: in "test_add32": critical check (uint8_t)0xAF == header.header()[0] has failed [0xaf != 0x54][0;39;49m
./gr-digital/lib/qa_header_buffer.cc(132): [4;31;49mfatal error: in "test_add64": critical check (uint8_t)0xAF == header.header()[0] has failed [0xaf != 0x67][0;39;49m
./gr-digital/lib/qa_header_buffer.cc(193): [4;31;49mfatal error: in "test_add_many": critical check (uint8_t)0x01 == header.header()[0] has failed [0x1 != 0x67][0;39;49m

[1;31;49m*** 4 failures are detected in the test module "Master Test Suite"
[0;39;49m

        Start 176: qa_binary_slicer_fb
176/261 Test #176: qa_binary_slicer_fb ..........................   Passed    0.17 sec
        Start 177: qa_burst_shaper
177/261 Test #177: qa_burst_shaper ..............................   Passed    0.18 sec
        Start 178: qa_chunks_to_symbols
178/261 Test #178: qa_chunks_to_symbols .........................   Passed    0.18 sec
        Start 179: qa_clock_recovery_mm
179/261 Test #179: qa_clock_recovery_mm .........................   Passed    0.17 sec
        Start 180: qa_constellation
180/261 Test #180: qa_constellation .............................***Failed    0.16 sec
Segmentation fault

        Start 181: qa_constellation_decoder_cb
181/261 Test #181: qa_constellation_decoder_cb ..................   Passed    0.17 sec
        Start 182: qa_constellation_encoder_bc
182/261 Test #182: qa_constellation_encoder_bc ..................   Passed    0.18 sec
        Start 183: qa_constellation_receiver
183/261 Test #183: qa_constellation_receiver ....................   Passed    0.81 sec
        Start 184: qa_constellation_soft_decoder_cf
184/261 Test #184: qa_constellation_soft_decoder_cf .............   Passed    0.99 sec
        Start 185: qa_correlate_access_code
185/261 Test #185: qa_correlate_access_code .....................   Passed    0.17 sec
        Start 186: qa_correlate_access_code_XX_ts
186/261 Test #186: qa_correlate_access_code_XX_ts ...............   Passed    0.18 sec
        Start 187: qa_correlate_access_code_tag
187/261 Test #187: qa_correlate_access_code_tag .................   Passed    0.18 sec
        Start 188: qa_costas_loop_cc
188/261 Test #188: qa_costas_loop_cc ............................   Passed    0.17 sec
        Start 189: qa_cpm
189/261 Test #189: qa_cpm .......................................   Passed    0.17 sec
        Start 190: qa_crc
190/261 Test #190: qa_crc .......................................   Passed    0.18 sec
        Start 191: qa_crc16_async_bb
191/261 Test #191: qa_crc16_async_bb ............................   Passed    0.16 sec
        Start 192: qa_crc32_async_bb
192/261 Test #192: qa_crc32_async_bb ............................***Failed    0.16 sec
F
======================================================================
FAIL: test_crc32 (__main__.qa_crc32_async_bb)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<<PKGBUILDDIR>>/gr-digital/python/digital/qa_crc32_async_bb.py", line 46, in test_crc32
    self.assertEqual(out_append, data + [0x88, 0xe2, 0xce, 0xce])
AssertionError: Lists differ: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 206, 206, 226, 136] != [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 136, 226, 206, 206]

First differing element 16:
206
136

- [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 206, 206, 226, 136]
?                                                                ----------

+ [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 136, 226, 206, 206]
?                                                       ++++++++++


----------------------------------------------------------------------
Ran 1 test in 0.005s

FAILED (failures=1)

        Start 193: qa_crc32_bb
193/261 Test #193: qa_crc32_bb ..................................***Failed    0.21 sec
...F.F.F....
======================================================================
FAIL: test_002_crc_equal_unpacked (__main__.qa_crc32_bb)
Test unpacked operation with packed operation
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<<PKGBUILDDIR>>/gr-digital/python/digital/qa_crc32_bb.py", line 193, in test_002_crc_equal_unpacked
    self.assertEqual(sink1.data(), sink2.data())
AssertionError: Lists differ: [[0, [211 chars], 0, 0, 1, 1, 1, 1, 0, 1, 1, 0, 0, 0, 0, 1, 1,[47 chars], 0]] != [[0, [211 chars], 0, 1, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 1,[47 chars], 1]]

First differing element 0:
[0, 0[210 chars], 0, 0, 1, 1, 1, 1, 0, 1, 1, 0, 0, 0, 0, 1, 1,[46 chars]0, 0]
[0, 0[210 chars], 0, 1, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 1,[46 chars]0, 1]

Diff is 898 characters long. Set self.maxDiff to None to see it.

======================================================================
FAIL: test_003_crc_equal_unpacked (__main__.qa_crc32_bb)
Test unpacked operation with packed operation
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<<PKGBUILDDIR>>/gr-digital/python/digital/qa_crc32_bb.py", line 218, in test_003_crc_equal_unpacked
    self.assertEqual(sink1.data(), sink2.data())
AssertionError: Lists differ: [[0, [835 chars], 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1,[47 chars], 1]] != [[0, [835 chars], 1, 1, 1, 0, 1, 1, 0, 1, 0, 0, 1, 0, 0, 0, 0,[47 chars], 0]]

First differing element 0:
[0, 0[834 chars], 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1,[46 chars]0, 1]
[0, 0[834 chars], 1, 1, 1, 0, 1, 1, 0, 1, 0, 0, 1, 0, 0, 0, 0,[46 chars]0, 0]

Diff is 2599 characters long. Set self.maxDiff to None to see it.

======================================================================
FAIL: test_005_tag_propagation (__main__.qa_crc32_bb)
Make sure tags on the CRC aren't lost.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<<PKGBUILDDIR>>/gr-digital/python/digital/qa_crc32_bb.py", line 131, in test_005_tag_propagation
    self.assertEqual([len(data) - 5, ], [
AssertionError: Lists differ: [8] != []

First list contains 1 additional elements.
First extra element 0:
8

- [8]
?  -

+ []

----------------------------------------------------------------------
Ran 12 tests in 0.043s

FAILED (failures=3)

        Start 194: qa_decision_feedback_equalizer
194/261 Test #194: qa_decision_feedback_equalizer ...............***Failed    0.17 sec
Segmentation fault

        Start 195: qa_diff_encoder
195/261 Test #195: qa_diff_encoder ..............................   Passed    0.25 sec
        Start 196: qa_diff_encoder_nrzi
196/261 Test #196: qa_diff_encoder_nrzi .........................   Passed    0.17 sec
        Start 197: qa_diff_phasor_cc
197/261 Test #197: qa_diff_phasor_cc ............................   Passed    0.17 sec
        Start 198: qa_digital
198/261 Test #198: qa_digital ...................................   Passed    0.16 sec
        Start 199: qa_fll_band_edge
199/261 Test #199: qa_fll_band_edge .............................   Passed    0.17 sec
        Start 200: qa_framer_sink
200/261 Test #200: qa_framer_sink ...............................   Passed    0.17 sec
        Start 201: qa_glfsr_source
201/261 Test #201: qa_glfsr_source ..............................   Passed    0.50 sec
        Start 202: qa_hdlc_framer
202/261 Test #202: qa_hdlc_framer ...............................   Passed    0.37 sec
        Start 203: qa_header_payload_demux
203/261 Test #203: qa_header_payload_demux ......................   Passed    3.11 sec
        Start 204: qa_lfsr
204/261 Test #204: qa_lfsr ......................................   Passed    0.17 sec
        Start 205: qa_linear_equalizer
205/261 Test #205: qa_linear_equalizer ..........................***Failed    0.16 sec
Segmentation fault

        Start 206: qa_map
206/261 Test #206: qa_map .......................................   Passed    0.17 sec
        Start 207: qa_meas_evm_cc
207/261 Test #207: qa_meas_evm_cc ...............................***Failed    0.18 sec
...Segmentation fault

        Start 208: qa_mpsk_snr_est
208/261 Test #208: qa_mpsk_snr_est ..............................   Passed    0.30 sec
        Start 209: qa_ofdm_carrier_allocator_cvc
209/261 Test #209: qa_ofdm_carrier_allocator_cvc ................   Passed    0.18 sec
        Start 210: qa_ofdm_chanest_vcvc
210/261 Test #210: qa_ofdm_chanest_vcvc .........................   Passed    0.21 sec
        Start 211: qa_ofdm_cyclic_prefixer
211/261 Test #211: qa_ofdm_cyclic_prefixer ......................   Passed    0.18 sec
        Start 212: qa_ofdm_frame_equalizer_vcvc
212/261 Test #212: qa_ofdm_frame_equalizer_vcvc .................   Passed    0.17 sec
        Start 213: qa_ofdm_serializer_vcc
213/261 Test #213: qa_ofdm_serializer_vcc .......................   Passed    0.17 sec
        Start 214: qa_ofdm_sync_sc_cfb
214/261 Test #214: qa_ofdm_sync_sc_cfb ..........................   Passed    0.24 sec
        Start 215: qa_ofdm_txrx
215/261 Test #215: qa_ofdm_txrx .................................   Passed    0.24 sec
        Start 216: qa_packet_format
216/261 Test #216: qa_packet_format .............................***Failed    0.48 sec
FF.
======================================================================
FAIL: test_packet_format_async_counter (__main__.test_packet_format_fb)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<<PKGBUILDDIR>>/gr-digital/python/digital/qa_packet_format.py", line 162, in test_packet_format_async_counter
    self.assertEqual(access_code, rx_access_code)
AssertionError: b'\xac\xdd\xa4\xe2\xf2\x8c \xfc' != b'\xfc \x8c\xf2\xe2\xa4\xdd\xac'

======================================================================
FAIL: test_packet_format_async_default (__main__.test_packet_format_fb)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<<PKGBUILDDIR>>/gr-digital/python/digital/qa_packet_format.py", line 68, in test_packet_format_async_default
    self.assertEqual(access_code, rx_access_code)
AssertionError: b'\xac\xdd\xa4\xe2\xf2\x8c \xfc' != b'\xfc \x8c\xf2\xe2\xa4\xdd\xac'

----------------------------------------------------------------------
Ran 3 tests in 0.309s

FAILED (failures=2)

        Start 217: qa_packet_headergenerator_bb
217/261 Test #217: qa_packet_headergenerator_bb .................   Passed    0.17 sec
        Start 218: qa_packet_headerparser_b
218/261 Test #218: qa_packet_headerparser_b .....................   Passed    0.98 sec
        Start 219: qa_pfb_clock_sync
219/261 Test #219: qa_pfb_clock_sync ............................   Passed    0.42 sec
        Start 220: qa_pn_correlator_cc
220/261 Test #220: qa_pn_correlator_cc ..........................   Passed    0.18 sec
        Start 221: qa_probe_density
221/261 Test #221: qa_probe_density .............................   Passed    0.17 sec
        Start 222: qa_scrambler
222/261 Test #222: qa_scrambler .................................   Passed    0.18 sec
        Start 223: qa_dtv
223/261 Test #223: qa_dtv .......................................***Failed    0.25 sec
Segmentation fault

        Start 224: qa_channel_model
224/261 Test #224: qa_channel_model .............................   Passed    0.17 sec
        Start 225: qa_fading_model
225/261 Test #225: qa_fading_model ..............................   Passed    0.16 sec
        Start 226: qa_add_system_time
226/261 Test #226: qa_add_system_time ...........................   Passed    1.17 sec
        Start 227: qa_pdu
227/261 Test #227: qa_pdu .......................................   Passed    0.36 sec
        Start 228: qa_pdu_lambda
228/261 Test #228: qa_pdu_lambda ................................   Passed    0.12 sec
        Start 229: qa_pdu_split
229/261 Test #229: qa_pdu_split .................................   Passed    0.16 sec
        Start 230: qa_pdu_to_stream
230/261 Test #230: qa_pdu_to_stream .............................   Passed    0.18 sec
        Start 231: qa_tags_to_pdu
231/261 Test #231: qa_tags_to_pdu ...............................   Passed    0.15 sec
        Start 232: qa_take_skip_to_pdu
232/261 Test #232: qa_take_skip_to_pdu ..........................   Passed    0.15 sec
        Start 233: qa_time_delta
233/261 Test #233: qa_time_delta ................................   Passed    0.17 sec
        Start 234: qa_iio
234/261 Test #234: qa_iio .......................................   Passed    1.69 sec
        Start 235: qa_qtgui
235/261 Test #235: qa_qtgui .....................................   Passed    1.16 sec
        Start 236: qa_trellis
236/261 Test #236: qa_trellis ...................................   Passed    0.24 sec
        Start 237: qa_uhd
237/261 Test #237: qa_uhd .......................................   Passed    0.34 sec
        Start 238: test_modtool
238/261 Test #238: test_modtool .................................   Passed    0.72 sec
        Start 239: qa_video_sdl
239/261 Test #239: qa_video_sdl .................................   Passed    0.19 sec
        Start 240: qa_alaw_vocoder
240/261 Test #240: qa_alaw_vocoder ..............................   Passed    0.18 sec
        Start 241: qa_cvsd_vocoder
241/261 Test #241: qa_cvsd_vocoder ..............................   Passed    0.16 sec
        Start 242: qa_g721_vocoder
242/261 Test #242: qa_g721_vocoder ..............................   Passed    0.16 sec
        Start 243: qa_g723_24_vocoder
243/261 Test #243: qa_g723_24_vocoder ...........................   Passed    0.15 sec
        Start 244: qa_g723_40_vocoder
244/261 Test #244: qa_g723_40_vocoder ...........................   Passed    0.16 sec
        Start 245: qa_ulaw_vocoder
245/261 Test #245: qa_ulaw_vocoder ..............................   Passed    0.16 sec
        Start 246: qa_codec2_vocoder
246/261 Test #246: qa_codec2_vocoder ............................   Passed    0.17 sec
        Start 247: qa_gsm_full_rate
247/261 Test #247: qa_gsm_full_rate .............................   Passed    0.15 sec
        Start 248: qa_classify
248/261 Test #248: qa_classify ..................................   Passed    0.18 sec
        Start 249: qa_zeromq_pub
249/261 Test #249: qa_zeromq_pub ................................   Passed    0.74 sec
        Start 250: qa_zeromq_pubsub
250/261 Test #250: qa_zeromq_pubsub .............................   Passed    4.18 sec
        Start 251: qa_zeromq_pull_msg_source
251/261 Test #251: qa_zeromq_pull_msg_source ....................   Passed    0.67 sec
        Start 252: qa_zeromq_pushpull
252/261 Test #252: qa_zeromq_pushpull ...........................   Passed    1.17 sec
        Start 253: qa_zeromq_req_msg_source
253/261 Test #253: qa_zeromq_req_msg_source .....................   Passed    0.67 sec
        Start 254: qa_zeromq_reqrep
254/261 Test #254: qa_zeromq_reqrep .............................   Passed    1.17 sec
        Start 255: qa_zeromq_sub
255/261 Test #255: qa_zeromq_sub ................................   Passed    6.38 sec
        Start 256: qa_zeromq_sub_msg_source
256/261 Test #256: qa_zeromq_sub_msg_source .....................   Passed    0.67 sec
        Start 257: qa_socket_pdu
257/261 Test #257: qa_socket_pdu ................................   Passed    2.66 sec
        Start 258: qa_tcp_sink
258/261 Test #258: qa_tcp_sink ..................................   Passed    0.45 sec
        Start 259: qa_udp_sink
259/261 Test #259: qa_udp_sink ..................................   Passed    0.45 sec
        Start 260: qa_udp_source
260/261 Test #260: qa_udp_source ................................   Passed    0.44 sec
        Start 261: qa_soapy_types
261/261 Test #261: qa_soapy_types ...............................   Passed    0.15 sec

93% tests passed, 18 tests failed out of 261

Total Test time (real) = 108.14 sec

The following tests FAILED:
	 20 - qa_pmt (Failed)
	 28 - blocks_qa_rotator.cc (Failed)
	 30 - qa_add_mult_div_sub (Failed)
	 65 - qa_multiply_matrix_xx (Failed)
	 82 - qa_rotator_cc (Failed)
	106 - qa_wavfile (Failed)
	141 - qa_freq_xlating_fft_filter (Failed)
	142 - qa_freq_xlating_fir_filter (Failed)
	174 - digital_qa_header_format.cc (Failed)
	175 - digital_qa_header_buffer.cc (Failed)
	180 - qa_constellation (Failed)
	192 - qa_crc32_async_bb (Failed)
	193 - qa_crc32_bb (Failed)
	194 - qa_decision_feedback_equalizer (Failed)
	205 - qa_linear_equalizer (Failed)
	207 - qa_meas_evm_cc (Failed)
	216 - qa_packet_format (Failed)
	223 - qa_dtv (Failed)
```

--- a/gr-blocks/include/gnuradio/blocks/rotator.h
+++ b/gr-blocks/include/gnuradio/blocks/rotator.h
@@ -49,7 +49,11 @@ public:
 
     void rotateN(gr_complex* out, const gr_complex* in, int n)
     {
+#if VOLK_VERSION >= 030100
+        volk_32fc_s32fc_x2_rotator2_32fc(out, in, &d_phase_incr, &d_phase, n);
+#else
         volk_32fc_s32fc_x2_rotator_32fc(out, in, d_phase_incr, &d_phase, n);
+#endif
     }
 };
 

--- a/gr-blocks/lib/multiply_by_tag_value_cc_impl.cc
+++ b/gr-blocks/lib/multiply_by_tag_value_cc_impl.cc
@@ -59,7 +59,11 @@ int multiply_by_tag_value_cc_impl::work(int noutput_items,
         end *= d_vlen;
 
         // Multiply based on the current value of k from 'start' to 'end'
+#if VOLK_VERSION >= 030100
+        volk_32fc_s32fc_multiply2_32fc(&out[start], &in[start], &d_k, (end - start));
+#else
         volk_32fc_s32fc_multiply_32fc(&out[start], &in[start], d_k, (end - start));
+#endif
         start = end;
 
         // Extract new value of k
@@ -75,8 +79,13 @@ int multiply_by_tag_value_cc_impl::work(int noutput_items,
         }
     }
 
+#if VOLK_VERSION >= 030100
+    volk_32fc_s32fc_multiply2_32fc(
+        &out[start], &in[start], &d_k, (d_vlen * noutput_items - start));
+#else
     volk_32fc_s32fc_multiply_32fc(
         &out[start], &in[start], d_k, (d_vlen * noutput_items - start));
+#endif
 
     return noutput_items;
 }

--- a/gr-blocks/lib/multiply_const_impl.cc
+++ b/gr-blocks/lib/multiply_const_impl.cc
@@ -73,7 +73,11 @@ int multiply_const_impl<gr_complex>::work(int noutput_items,
     gr_complex* out = (gr_complex*)output_items[0];
     int noi = noutput_items * d_vlen;
 
+#if VOLK_VERSION >= 030100
+    volk_32fc_s32fc_multiply2_32fc(out, in, &d_k, noi);
+#else
     volk_32fc_s32fc_multiply_32fc(out, in, d_k, noi);
+#endif
 
     return noutput_items;
 }

--- a/gr-blocks/lib/multiply_matrix_impl.cc
+++ b/gr-blocks/lib/multiply_matrix_impl.cc
@@ -262,7 +262,11 @@ int multiply_matrix_impl<gr_complex>::work(int noutput_items,
         gr_complex* out = reinterpret_cast<gr_complex*>(output_items[out_idx]);
         // Do input 0 first, this saves a memset
         const gr_complex* in = reinterpret_cast<const gr_complex*>(input_items[0]);
+#if VOLK_VERSION >= 030100
+        volk_32fc_s32fc_multiply2_32fc(out, in, &d_A[out_idx][0], noutput_items);
+#else
         volk_32fc_s32fc_multiply_32fc(out, in, d_A[out_idx][0], noutput_items);
+#endif
         // Then do inputs 1 through N
         for (size_t in_idx = 1; in_idx < input_items.size(); in_idx++) {
             in = reinterpret_cast<const gr_complex*>(input_items[in_idx]);

--- a/gr-blocks/lib/phase_shift_impl.cc
+++ b/gr-blocks/lib/phase_shift_impl.cc
@@ -84,7 +84,11 @@ int phase_shift_impl::work(int noutput_items,
     gr::thread::scoped_lock guard(d_setlock);
 
     if (d_shift != 0.0f) {
+#if VOLK_VERSION >= 030100
+        volk_32fc_s32fc_multiply2_32fc(out, in, &d_shift_cc, noutput_items);
+#else
         volk_32fc_s32fc_multiply_32fc(out, in, d_shift_cc, noutput_items);
+#endif
     } else {
         memcpy(out, in, sizeof(gr_complex) * noutput_items);
     }

--- a/gr-blocks/python/blocks/bindings/rotator_python.cc
+++ b/gr-blocks/python/blocks/bindings/rotator_python.cc
@@ -14,7 +14,7 @@
 /* BINDTOOL_GEN_AUTOMATIC(0)                                                       */
 /* BINDTOOL_USE_PYGCCXML(0)                                                        */
 /* BINDTOOL_HEADER_FILE(rotator.h)                                        */
-/* BINDTOOL_HEADER_FILE_HASH(5d52f019c659d77040b16ad7541e4fd2)                     */
+/* BINDTOOL_HEADER_FILE_HASH(3af403c5adb7af31a51113901293189d)                     */
 /***********************************************************************************/
 
 #include <pybind11/complex.h>

--- a/gr-digital/include/gnuradio/digital/adaptive_algorithm_cma.h
+++ b/gr-digital/include/gnuradio/digital/adaptive_algorithm_cma.h
@@ -70,8 +70,13 @@ public:
         gr_complex err_x_mu = -d_step_size * error;
 
         volk_32fc_conjugate_32fc(conj_vector.data(), in, num_taps);
+#if VOLK_VERSION >= 030100
+        volk_32fc_s32fc_multiply2_32fc(
+            prod_vector.data(), conj_vector.data(), &err_x_mu, num_taps);
+#else
         volk_32fc_s32fc_multiply_32fc(
             prod_vector.data(), conj_vector.data(), err_x_mu, num_taps);
+#endif
         volk_32fc_x2_add_32fc(taps, taps, prod_vector.data(), num_taps);
     }
 

--- a/gr-digital/include/gnuradio/digital/adaptive_algorithm_lms.h
+++ b/gr-digital/include/gnuradio/digital/adaptive_algorithm_lms.h
@@ -51,8 +51,13 @@ public:
         gr_complex err_x_mu = d_step_size * error;
 
         volk_32fc_conjugate_32fc(conj_vector.data(), in, num_taps);
+#if VOLK_VERSION >= 030100
+        volk_32fc_s32fc_multiply2_32fc(
+            prod_vector.data(), conj_vector.data(), &err_x_mu, num_taps);
+#else
         volk_32fc_s32fc_multiply_32fc(
             prod_vector.data(), conj_vector.data(), err_x_mu, num_taps);
+#endif
         volk_32fc_x2_add_32fc(taps, taps, prod_vector.data(), num_taps);
     }
 

--- a/gr-digital/python/digital/bindings/adaptive_algorithm_cma_python.cc
+++ b/gr-digital/python/digital/bindings/adaptive_algorithm_cma_python.cc
@@ -14,7 +14,7 @@
 /* BINDTOOL_GEN_AUTOMATIC(0)                                                       */
 /* BINDTOOL_USE_PYGCCXML(0)                                                        */
 /* BINDTOOL_HEADER_FILE(adaptive_algorithm_cma.h) */
-/* BINDTOOL_HEADER_FILE_HASH(db6db4f2b808db300412b90c61b333b6)                     */
+/* BINDTOOL_HEADER_FILE_HASH(0b4e0edcded1314dd33e1733e867be2f)                     */
 /***********************************************************************************/
 
 #include <pybind11/complex.h>

--- a/gr-digital/python/digital/bindings/adaptive_algorithm_lms_python.cc
+++ b/gr-digital/python/digital/bindings/adaptive_algorithm_lms_python.cc
@@ -14,7 +14,7 @@
 /* BINDTOOL_GEN_AUTOMATIC(0)                                                       */
 /* BINDTOOL_USE_PYGCCXML(0)                                                        */
 /* BINDTOOL_HEADER_FILE(adaptive_algorithm_lms.h) */
-/* BINDTOOL_HEADER_FILE_HASH(f3725cfd42cb5687fe2b2191b78e96b6)                     */
+/* BINDTOOL_HEADER_FILE_HASH(bf34ea7730349513a808de7f99e7744b)                     */
 /***********************************************************************************/
 
 #include <pybind11/complex.h>

--- a/gr-dtv/lib/dvbt/dvbt_reference_signals_impl.cc
+++ b/gr-dtv/lib/dvbt/dvbt_reference_signals_impl.cc
@@ -14,6 +14,7 @@
 #include <gnuradio/expj.h>
 #include <gnuradio/io_signature.h>
 #include <gnuradio/math.h>
+#include <volk/volk.h>
 #include <algorithm>
 #include <complex>
 
@@ -1077,8 +1078,10 @@ int dvbt_reference_signals_impl::general_work(int noutput_items,
                &out[(i * d_noutput) + (ofdm_fft_size / 2)],
                sizeof(gr_complex) * ofdm_fft_size / 2);
         ofdm_fft.execute();
-        volk_32fc_s32fc_multiply_32fc(
-            &out[i * d_noutput], ofdm_fft.get_outbuf(), normalization, ofdm_fft_size);
+        volk_32f_s32f_multiply_32f(reinterpret_cast<float*>(&out[i * d_noutput]),
+                                   reinterpret_cast<float*>(ofdm_fft.get_outbuf()),
+                                   normalization,
+                                   ofdm_fft_size * 2);
     }
 
     // Tell runtime system how many input items we consumed on

--- a/gr-dtv/lib/dvbt2/dvbt2_paprtr_cc_impl.cc
+++ b/gr-dtv/lib/dvbt2/dvbt2_paprtr_cc_impl.cc
@@ -671,8 +671,10 @@ int dvbt2_paprtr_cc_impl::work(int noutput_items,
                     memcpy(ones_time.data(),
                            papr_fft.get_outbuf(),
                            sizeof(gr_complex) * papr_fft_size);
-                    volk_32fc_s32fc_multiply_32fc(
-                        ones_time.data(), ones_time.data(), normalization, papr_fft_size);
+                    volk_32f_s32f_multiply_32f(reinterpret_cast<float*>(ones_time.data()),
+                                               reinterpret_cast<float*>(ones_time.data()),
+                                               normalization,
+                                               papr_fft_size * 2);
                     std::fill_n(&r[0], N_TR, 0);
                     std::fill_n(&c[0], papr_fft_size, 0);
                     for (int k = 1; k <= num_iterations; k++) {
@@ -706,7 +708,11 @@ int dvbt2_paprtr_cc_impl::work(int noutput_items,
                                 papr_fft_size;
                             ctemp[n] = std::exp(gr_complexd(0.0, vtemp));
                         }
+#if VOLK_VERSION >= 030100
+                        volk_32fc_s32fc_multiply2_32fc(v.data(), ctemp.data(), &u, N_TR);
+#else
                         volk_32fc_s32fc_multiply_32fc(v.data(), ctemp.data(), u, N_TR);
+#endif
                         volk_32f_s32f_multiply_32f(
                             (float*)rNew.data(), (float*)v.data(), alpha, N_TR * 2);
                         volk_32f_x2_subtract_32f((float*)rNew.data(),
@@ -746,8 +752,13 @@ int dvbt2_paprtr_cc_impl::work(int noutput_items,
                             ones_freq[(n + m) % papr_fft_size] = ones_time[n];
                         }
                         result = u * alpha;
+#if VOLK_VERSION >= 030100
+                        volk_32fc_s32fc_multiply2_32fc(
+                            ctemp.data(), ones_freq.data(), &result, papr_fft_size);
+#else
                         volk_32fc_s32fc_multiply_32fc(
                             ctemp.data(), ones_freq.data(), result, papr_fft_size);
+#endif
                         volk_32f_x2_subtract_32f((float*)c.data(),
                                                  (float*)c.data(),
                                                  (float*)ctemp.data(),

--- a/gr-dtv/lib/dvbt2/dvbt2_pilotgenerator_cc_impl.cc
+++ b/gr-dtv/lib/dvbt2/dvbt2_pilotgenerator_cc_impl.cc
@@ -2701,8 +2701,10 @@ int dvbt2_pilotgenerator_cc_impl::general_work(int noutput_items,
             memcpy(
                 &dst[0], &out[ofdm_fft_size / 2], sizeof(gr_complex) * ofdm_fft_size / 2);
             ofdm_fft.execute();
-            volk_32fc_s32fc_multiply_32fc(
-                out, ofdm_fft.get_outbuf(), normalization, ofdm_fft_size);
+            volk_32f_s32f_multiply_32f(reinterpret_cast<float*>(out),
+                                       reinterpret_cast<float*>(ofdm_fft.get_outbuf()),
+                                       normalization,
+                                       ofdm_fft_size * 2);
             out += ofdm_fft_size;
         }
     }

